client/src/components/ApplicationForm/Dropzone.tsx

```diff
@@ -0,0 +1,44 @@
+import { useCallback, useState, Dispatch, SetStateAction } from 'react';
+import { useDropzone } from 'react-dropzone';
+
+type DropzoneProps = {
+  className: string;
+  setFile: Dispatch<SetStateAction<File | undefined>>;
```

Comment:
I find it rather ugly to use `Dispatch` in my callback props. I prefer to be explicit about what I will pass it. In this case, it looks like you could declare this as:
`setFile: (file: File) => void;`
(It doesn't look like you ever pass it `undefined`, so don't declare that it takes undefined, which will make it easier for TS to do better error checking for you, and will make your coding easier.)